Specify priority for VMs in Azure VM scale sets

arvindshmicrosoft opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Adds support to create VMSS with low-priority VMs through a new optional
parameter. The default is to create a VMSS with Regular priority VMs.

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Feature Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
azure_rm_virtualmachine_scaleset

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
https://docs.microsoft.com/en-us/azure/virtual-machine-scale-sets/virtual-machine-scale-sets-use-low-priority

Fred-sun commented:
ready_for_review

Fred-sun commented:
ready_for_review

Fred-sun commented:
@arvindshmicrosoft Thanks for your contribution, Would you please help resolve the conflicting files ? When you finish change, I will push for view again! Thanks a lot!

arvindshmicrosoft commented:
@Fred-sun done. Thanks for following up, I appreciate that! In addition, please let me know what I can do to help expedite the merge process - I have a couple of PRs which are pending review and merge...

Fred-sun commented:
> @Fred-sun done. Thanks for following up, I appreciate that! In addition, please let me know what I can do to help expedite the merge process - I have a couple of PRs which are pending review and merge...

@arvindshmicrosoft Thanks for your update, I will ping committer help to review!

Fred-sun commented:
@haiyuazhang @gavinfish review_need

Fred-sun commented:
@arvindshmicrosoft  Please help resolve the conflicting files when you are free! will push for PR review and merger! Thank you very much!

arvindshmicrosoft committed:
Specify priority for VMs in Azure VM scale sets
Adds support to create VMSS with low-priority VMs through a new optional
parameter. The default is to create a VMSS with Regular priority VMs.
https://docs.microsoft.com/en-us/azure/virtual-machine-scale-sets/virtual-machine-scale-sets-use-low-priority
for details.

arvindshmicrosoft commented:
@Fred-sun I have done the needful. I'm sure you also know that it has been over 2 months since my original PR and since then this is the 2nd time I have had to rebase due to other conflicting changes from other contributions which arrived later. I'm sure you agree that this is not an optimal experience for anyone. Please see what you can do to see this through. Thanks in advance.

gavinfish reviewed (approved):
shipit
Ref https://docs.microsoft.com/en-us/azure/templates/microsoft.compute/2018-10-01/virtualmachinescalesets.

Fred-sun commented:
> @Fred-sun I have done the needful. I'm sure you also know that it has been over 2 months since my original PR and since then this is the 2nd time I have had to rebase due to other conflicting changes from other contributions which arrived later. I'm sure you agree that this is not an optimal experience for anyone. Please see what you can do to see this through. Thanks in advance.

Thank you for your contribution, because recently there are many things in the community, resulting in the negligence of some problems. It was our fault. I will respond positively and follow up these problems in the future. Thanks again for your reminding.

haiyuazhang reviewed (approved):
shipit
Ref https://docs.microsoft.com/en-us/azure/templates/microsoft.compute/2018-10-01/virtualmachinescalesets.
